### ISSUE TITLE
Random failed tests in TestIsGte and TestIsLt

### ISSUE THREAD

**Issue by zhsj** [go-playground/validator#294]
### Package version eg. v8, v9: 

both v8 and v9.

### Issue, Question or Enhancement:

```
cloud@ustc1 ~/go-test/src/gopkg.in/go-playground/validator.v8 (master) $ go test ./... -run TestIsGte
ok      gopkg.in/go-playground/validator.v8     0.005s
?       gopkg.in/go-playground/validator.v8/examples/custom     [no test files]
?       gopkg.in/go-playground/validator.v8/examples/simple     [no test files]
?       gopkg.in/go-playground/validator.v8/examples/struct-level       [no test files]
cloud@ustc1 ~/go-test/src/gopkg.in/go-playground/validator.v8 (master) $ go test ./... -run TestIsGte
ok      gopkg.in/go-playground/validator.v8     0.005s
?       gopkg.in/go-playground/validator.v8/examples/custom     [no test files]
?       gopkg.in/go-playground/validator.v8/examples/simple     [no test files]
?       gopkg.in/go-playground/validator.v8/examples/struct-level       [no test files]
cloud@ustc1 ~/go-test/src/gopkg.in/go-playground/validator.v8 (master) $ go test ./... -run TestIsGte
ok      gopkg.in/go-playground/validator.v8     0.005s
?       gopkg.in/go-playground/validator.v8/examples/custom     [no test files]
?       gopkg.in/go-playground/validator.v8/examples/simple     [no test files]
?       gopkg.in/go-playground/validator.v8/examples/struct-level       [no test files]
cloud@ustc1 ~/go-test/src/gopkg.in/go-playground/validator.v8 (master) $ go test ./... -run TestIsGte
ok      gopkg.in/go-playground/validator.v8     0.005s
?       gopkg.in/go-playground/validator.v8/examples/custom     [no test files]
?       gopkg.in/go-playground/validator.v8/examples/simple     [no test files]
?       gopkg.in/go-playground/validator.v8/examples/struct-level       [no test files]
cloud@ustc1 ~/go-test/src/gopkg.in/go-playground/validator.v8 (master) $ go test ./... -run TestIsGte
validator_test.go:4881 <nil> should not be equal <nil>
--- FAIL: TestIsGte (0.00s)
FAIL
FAIL    gopkg.in/go-playground/validator.v8     0.006s
?       gopkg.in/go-playground/validator.v8/examples/custom     [no test files]
?       gopkg.in/go-playground/validator.v8/examples/simple     [no test files]
?       gopkg.in/go-playground/validator.v8/examples/struct-level       [no test files]




cloud@ustc1 ~/go-test/src/gopkg.in/go-playground/validator.v9 (master) $ go test ./... -run TestIsGte
ok      gopkg.in/go-playground/validator.v9     0.006s
ok      gopkg.in/go-playground/validator.v9/translations/en     0.010s [no tests to run]
cloud@ustc1 ~/go-test/src/gopkg.in/go-playground/validator.v9 (master) $ go test ./... -run TestIsGte
validator_test.go:5282 <nil> should not be equal <nil>
--- FAIL: TestIsGte (0.00s)
FAIL
FAIL    gopkg.in/go-playground/validator.v9     0.007s
ok      gopkg.in/go-playground/validator.v9/translations/en     0.004s [no tests to run]
```

IMHO, that's caused by same time.Now() result when the two calls are very close.

![image](https://user-images.githubusercontent.com/227180/28580592-81d94900-7192-11e7-9051-f966e4fd1b1b.png)

**Comment by joeybloggs** [comment]
Tanks @zhsj you're probably right, I'll adjust the tests to remove the false positive; nice that it's running so fast though!

**Comment by joeybloggs** [comment]
This should be fixed in [v9 release v9.5.0](https://github.com/go-playground/validator/releases/tag/v9.5.0)

don't know if i'll go back to fix in v8, I highly recommend upgrading to v9 or v10 which is on the horizon.

**Comment by zhsj** [comment]
Thanks. 

We are packaging this library inside Debian[1][2], as it's the dependency of the current version of github.com/gin-gonic/gin.  I will cherry pick your fix to v8 if you don't want to. But wish you can fix it in v8 too.

[1] https://ftp-master.debian.org/new/golang-gopkg-go-playground-validator.v8_8.18.1-1.html
[2] https://bugs.debian.org/869369

**Comment by joeybloggs** [comment]
Ok considering that I'll backport the fix for you tonight or tomorrow.

**Comment by joeybloggs** [comment]
OK @zhsj I've backported the fixes to v8 in [release 8.18.12](https://github.com/go-playground/validator/releases/tag/v8.18.2)

please let me know if you encounter the issue again or need anything else :)

**Comment by zhsj** [comment]
Thanks! Close it now.
